tft.py

```diff
@@ -0,0 +1,213 @@
+
+                item_obj["effects"].append(effect_obj)
+
+            items[item_obj["id"]] = item_obj
+
+        for item in items:
+            # there must be some list comprehention way of doing this...
+
+            items[item]["newf"] = []
+            for old in items[item]["from"]:
+                key = [x for x in items if old in items[x]["internal"]["hex"]][0]
+                items[item]["newf"].append(key)
+
+            items[item]["from"] = items[item]["newf"]
+
+        return items
+
+    def parse_champs(self, map22, traits):
+        champ_collection = [x for x in map22.entries if x.type == "TftShopData"]
+        champs = {}
+
+        for champ in champ_collection:
+            name = champ.getv("mName")
+
+            if (not name.startswith("TFT_")) or name == "TFT_Template":
+                continue
+
+            char_bin = BinFile(os.path.join(self.character_folder, name[4:], name[4:] + ".bin"))
+            champ_id = [x.getv("characterToolData").getv("championId") for x in char_bin.entries if x.type == "CharacterRecord"][0]
+
+            champ_obj = {}
+            champ_obj["name"] = champ.getrv("c3143d66")
+            champ_obj["cost"] = champ.getv("mRarity") or 1
+            champ_obj["icon"] = champ.getv("mIconPath")
+            champ_obj["traits"] = []
+
+            tft_bin = BinFile(os.path.join(self.character_folder, name, name + ".bin"))
+            record = [x for x in tft_bin.entries if x.type == "TFTCharacterRecord"][0]
+            for trait in record.getv("mLinkedTraits"):
+                champ_obj["traits"].append([x["internal"]["name"] for x in traits if x["internal"]["hash"] == trait.hex][0])
+
+            champ_obj["stats"] = {}
+            champ_obj["stats"]["hp"] = record.getv("baseHP")
+            champ_obj["stats"]["mana"] = record.getv("primaryAbilityResource").getv("arBase")
+            champ_obj["stats"]["initialMana"] = record.getv("mInitialMana") or 0
+            champ_obj["stats"]["damage"] = record.getv("BaseDamage")
+            champ_obj["stats"]["armor"] = record.getv("baseArmor")
+            champ_obj["stats"]["magicResist"] = record.getv("baseSpellBlock")
+            champ_obj["stats"]["critMultiplier"] = record.getv("critDamageMultiplier")
+            champ_obj["stats"]["critChance"] = record.getv("baseCritChance")
+            champ_obj["stats"]["attackSpeed"] = record.getv("attackSpeed")
+            champ_obj["stats"]["range"] = record.getv("attackRange") // 180
+
+            champ_obj["ability"] = {}
+            champ_obj["ability"]["name"] = champ.getrv("87a69a5e")
+            champ_obj["ability"]["desc"] = champ.getrv("bc4f18b3")
+            champ_obj["ability"]["icon"] = champ.getv("mPortraitIconPath")
+            champ_obj["ability"]["variables"] = []
+
+            spellName = record.getv("spellNames")[0]
+            if "/" in spellName:
+                spellName = spellName[spellName.find("/") + 1 :]
```

Comment:
Do you want exactly the 2d part or the last one (the basename)?
If you want the basename, you can use `os.path.basename(spellName)`.

Reply:
2nd part